microservice/subscription/src/test/java/com/cumulocity/microservice/subscription/repository/impl/LegacyMicroserviceRepositoryTest.java

```diff
@@ -320,6 +322,86 @@ public void shouldGetSubscriptions(){
                 .isSameAs(user);
     }
 
+    @Test
+    public void shouldAddExtensionsToAppIfMetadataContains() {
+        repository.register(
+                microserviceMetadataRepresentation()
+                        .extensions(Arrays.asList(new ExtensionRepresentation()))
+                        .build());
+
+        assertThat(platform.take(byMethod(POST)))
+                .hasSize(1)
+                .extracting("body")
+                .have(new Condition<Object>() {
+                    @Override
+                    public boolean matches(Object value) {
+                        if (value instanceof ApplicationRepresentation) {
+                            assertThat(((ApplicationRepresentation) value).get(MicroserviceMetadataRepresentation.EXTENSIONS_FIELD_NAME)).isNotNull();
+                            return true;
+                        }
+                        return false;
+                    }
+                });
+        assertThat(platform.take(byMethod(PUT))).isEmpty();
```

Comment:
I don't really get how this test is working, the first `repository.register` is recording manifest with extensions only for the next call?
Can you make is more readable by extracting methods like `givenManifestWithExtensions` and maybe it make sense to split this one test into few different cases?

Reply:
I am checking here create and update case so doing another register call after this and FakeCredentialsSwitchingPlatform platform used in the test class is able to manage POST and PUT correctly. 

I can still split it and do more readable :+1: